### DISCUSSION TITLE
Hooks and metadata extensions

### DISCUSSION THREAD

**Discussion by sted** [juicedata/juicefs#5066] [General]
Hi all,
Does JuiceFS have the capability to extend node/file metadata? Does it support the concept of hooks for various file system operations?

Alternatively, what is your advice on creating relationships with records representing nodes?

**Comment by zhijian-pro** [comment]
No plan, any specific application scenarios?

**Reply by sted** [reply]
To extend the functionality of JuiceFS, for instance by adding workflow features or other capabilities, without the need to duplicate information. JuiceFS uses a metadata engine (such as PostgreSQL, Redis, etc.), which could be extended to offer additional features on top of its current capabilities.

**Reply by zhijian-pro** [reply]
For now, you need to implement it yourself manually
